scipy/sparse/linalg/tests/test_expm_multiply.py

```diff
@@ -160,36 +175,31 @@ def test_sparse_expm_multiply_interval(self):
 
     def test_expm_multiply_interval_vector(self):
         np.random.seed(1234)
-        start = 0.1
-        stop = 3.2
-        endpoint = True
-        for num in (14, 13, 2):
-            for n in (1, 2, 5, 20, 40):
-                A = scipy.linalg.inv(np.random.randn(n, n))
-                v = np.random.randn(n)
-                X = expm_multiply(A, v,
-                        start=start, stop=stop, num=num, endpoint=endpoint)
-                samples = np.linspace(start=start, stop=stop,
-                        num=num, endpoint=endpoint)
-                for solution, t in zip(X, samples):
-                    assert_allclose(solution, scipy.linalg.expm(t*A).dot(v))
+        interval = {'start': 0.1, 'stop': 3.2, 'endpoint': True}
+        for num, n in product([14, 13, 2], [1, 2, 5, 20, 40]):
+            A = scipy.linalg.inv(np.random.randn(n, n))
+            v = np.random.randn(n)
+            samples = np.linspace(num=num, **interval)
+            X = expm_multiply(A, v, num=num, **interval)
+            for solution, t in zip(X, samples):
+                assert_allclose(solution, scipy.linalg.expm(t*A).dot(v))
+            X = expm_multiply(aslinearoperator(A), v, num=num, **interval)
+            for solution, t in zip(X, samples):
+                assert_allclose(solution, scipy.linalg.expm(t*A).dot(v))
 
     def test_expm_multiply_interval_matrix(self):
         np.random.seed(1234)
-        start = 0.1
-        stop = 3.2
-        endpoint = True
-        for num in (14, 13, 2):
-            for n in (1, 2, 5, 20, 40):
-                for k in (1, 2):
-                    A = scipy.linalg.inv(np.random.randn(n, n))
-                    B = np.random.randn(n, k)
-                    X = expm_multiply(A, B,
-                            start=start, stop=stop, num=num, endpoint=endpoint)
-                    samples = np.linspace(start=start, stop=stop,
-                            num=num, endpoint=endpoint)
-                    for solution, t in zip(X, samples):
-                        assert_allclose(solution, scipy.linalg.expm(t*A).dot(B))
+        interval = {'start': 0.1, 'stop': 3.2, 'endpoint': True}
+        for num, n, k in product([14, 13, 2], [1, 2, 5, 20, 40], [1, 2]):
+            A = scipy.linalg.inv(np.random.randn(n, n))
+            B = np.random.randn(n, k)
+            samples = np.linspace(num=num, **interval)
+            X = expm_multiply(A, B, num=num, **interval)
+            for solution, t in zip(X, samples):
+                assert_allclose(solution, scipy.linalg.expm(t*A).dot(B))
+            X = expm_multiply(aslinearoperator(A), B, num=num, **interval)
+            for solution, t in zip(X, samples):
+                assert_allclose(solution, scipy.linalg.expm(t*A).dot(B))
```

Comment:
Is this just a refactor of the organization of the original test?

It looks like that on the surface, but for a non-linalg expert like me it does increase my nervousness about the potential of a test being changed when it isn't really related to the core changes in the PR.

Reply:
I needed to refactor the test, as now a line limit is enforced by the automatic linters. I did so by collapsing the nested for loops using `itertools.product`, as well as saving repeated key-word arguments in the dictionary `interval`.
No other changes should have been applied to the existing test, please complain if you disagree.
These changes should not involve any linear algebra, only Python syntax.

The second thing is, of-course, repeating the test treating `A` as a linear operator.

Reply:
@DerWeh Maybe consider adding tests for float32 and complex arithmetic using 
```
@pytest.mark.parametrize("dtype", (float, float32, complex))
``` 
like e.g. in https://github.com/scipy/scipy/blob/master/scipy/sparse/linalg/_eigen/tests/test_svds.py